rootston/keyboard.c

```diff
@@ -116,13 +125,36 @@ static void keyboard_key_notify(struct wl_listener *listener, void *data) {
 	const xkb_keysym_t *syms;
 	int syms_len = xkb_state_key_get_syms(keyboard->device->keyboard->xkb_state,
 		keycode, &syms);
+
+	bool handled = false;
 	for (int i = 0; i < syms_len; i++) {
 		if (event->state == WLR_KEY_PRESSED) {
-			keyboard_keysym_press(keyboard, syms[i]);
+			bool keysym_handled = keyboard_keysym_press(keyboard, syms[i]);
+			handled = handled || keysym_handled;
 		} else { // WLR_KEY_RELEASED
 			keyboard_keysym_release(keyboard, syms[i]);
 		}
 	}
+
+	if (!handled) {
+		wlr_seat_set_keyboard(keyboard->input->wl_seat, keyboard->device);
```

Comment:
We could get fancy here and have the seat read the keyboard event before rootston and automatically set the keyboard on the seat to the one with the last event then lazily send the keymap on the next seat key notify.

Reply:
Yeah, that would be better, the compositor should always do it anyway.